internal/token/token.go

```diff
@@ -0,0 +1,96 @@
+package token
+
+import (
+	"bufio"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type TokenClient interface {
```

Comment:
This interface and the way it is being used is kinda strange. You've got an interface, implying that there are potentially multiple implementations of it, but then various key codepaths call `NewFileTokenClient` internally. Also, `NewFileTokenClient` can actually fail.

A better approach would be to make the `GetToken/CheckToken` methods potentially return errors, and then drop the constructor for the `FileTokenClient` altogether. Instead, you'd initialize it like `&FileTokenClient{Path: path}` and any code that needs a client would have one passed in. This would delay the file check until `GetToken` is called, so that if there is a problem with that, other code paths won't fail prematurely.

Reply:
I rather thought that the fact that the constructor failed early and that the token file was read only once was a feature — but perhaps that's not the Go way. Will change.

Reply:
So I won't claim to be the arbiter of the one true "Go way".

My experience though has been that the fewer constructors/factory-functions, the better. Delaying errors makes things more dynamic, lazy-loaded, etc. too. So in this case, we won't fail on startup for like `exo help` or `exo version` or similar when we load config and try to create a client, but something goes wrong!